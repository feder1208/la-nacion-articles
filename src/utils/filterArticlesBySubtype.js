function filterArticlesBySubtype(articles = []) {
    return articles.filter((article) => article.subtype === "7");
}

export {filterArticlesBySubtype}