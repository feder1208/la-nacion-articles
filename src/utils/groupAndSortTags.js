function groupAndSortTags(articles = []) {
  const tags = articles
    .map((article) => {
      return article.taxonomy.tags;
    })
    .flat();

  function groupTags(array) {
    return array.reduce((acc, curr) => {
      const key = curr.text;

      if (!acc.some((position) => position.text === key)) {
        acc.push({ ...curr, count: 0 });
      }

      acc[acc.findIndex((el) => el.text === key)].count += 1;

      return acc;
    }, []);
  }

  let groupedTags = groupTags(tags);

  const sortedTags = groupedTags
    .sort((a, b) => b.count - a.count)
    .splice(0, 10);

  return sortedTags;
}

export { groupAndSortTags };
