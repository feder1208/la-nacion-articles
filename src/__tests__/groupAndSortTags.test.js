/* eslint-disable jest/no-mocks-import */
import { groupAndSortTags } from "../utils/groupAndSortTags";
import fetchDataMock from "../__mocks__/fetchDataMock.json";

describe("Group the tags in ascending order", () => {
    test("Should return 10 objects by count", () => {
        
      const tags = groupAndSortTags(fetchDataMock.articles);
      expect(tags).toHaveLength(10);
      expect(tags).toMatchSnapshot();
    });
  });