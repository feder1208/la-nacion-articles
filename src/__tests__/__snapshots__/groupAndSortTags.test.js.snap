// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group the tags in ascending order Should return 10 objects by count 1`] = `
Array [
  Object {
    "count": 13,
    "slug": "huevo-tid47236",
    "text": "Huevo",
  },
  Object {
    "count": 9,
    "slug": "leche-tid47244",
    "text": "Leche",
  },
  Object {
    "count": 7,
    "slug": "arroz-tid47136",
    "text": "Arroz",
  },
  Object {
    "count": 7,
    "slug": "manteca-tid47257",
    "text": "Manteca",
  },
  Object {
    "count": 7,
    "slug": "azucar-tid47141",
    "text": "Az√∫car",
  },
  Object {
    "count": 5,
    "slug": "harina-0000-tid48184",
    "text": "Harina 0000",
  },
  Object {
    "count": 4,
    "slug": "pescados-tid67216",
    "text": "Pescados",
  },
  Object {
    "count": 4,
    "slug": "ajo-tid47126",
    "text": "Ajo",
  },
  Object {
    "count": 4,
    "slug": "frutas-tid67217",
    "text": "Frutas",
  },
  Object {
    "count": 3,
    "slug": "canela-tid47164",
    "text": "Canela",
  },
]
`;
